packages/contractkit/src/identity/offchain/schemas/utils.ts

```diff
@@ -157,40 +205,45 @@ export const readEncrypted = async (
   return Ok(payload)
 }
 
-export const resolveEncrypted = async (
+/**
+ * Reads and decrypts a payload that has been encrypted to your data encryption key. Will
+ * resolving the symmetric key used to encrypt the payload.
+ *
+ * @param wrapper the offchain data wrapper
+ * @param dataPath path to where the encrypted data is stored. Used to derive the key location
+ * @param senderAddress the address that encrypted this key to you
+ */
+export const readEncrypted = async (
   wrapper: OffchainDataWrapper,
   dataPath: string,
   senderAddress: string
 ): Promise<Result<Buffer, SchemaErrors>> => {
   const encryptedPayloadPath = `${dataPath}.enc`
-  const [encryptedPayload, keyOrPayload] = await Promise.all([
+  const [payload, key] = await Promise.all([
     wrapper.readDataFromAsResult(
       senderAddress,
       (buf) => buildEIP712TypedData(wrapper, encryptedPayloadPath, buf),
       encryptedPayloadPath
     ),
-    readEncrypted(wrapper, dataPath, senderAddress),
+    readSymmetricKey(wrapper, dataPath, senderAddress),
   ])
 
-  if (!keyOrPayload.ok) {
-    return Err(keyOrPayload.error)
+  if (!payload.ok) {
+    return Err(new OffchainError(payload.error))
+  }
+  if (!key.ok) {
+    return Err(key.error)
   }
 
-  if (encryptedPayload.ok && keyOrPayload.ok) {
-    const key = keyOrPayload.result
-    if (key.length !== KEY_LENGTH) {
-      return Err(new InvalidKey())
-    }
-
-    const payload = encryptedPayload.result
-    const iv = payload.slice(0, IV_LENGTH)
-    const encryptedData = payload.slice(IV_LENGTH)
-    const decipher = createDecipheriv('aes-128-ctr', key, iv)
-
-    return Ok(Buffer.concat([decipher.update(encryptedData), decipher.final()]))
+  if (key.result.length !== KEY_LENGTH) {
+    return Err(new InvalidKey())
   }
 
-  return keyOrPayload
+  const iv = payload.result.slice(0, IV_LENGTH)
```

Comment:
Maybe separate the "cryptography" into a separate function, it seems like different layers of concerns here